### PR TITLE
Add knot

### PR DESCRIPTION
**Pull request by ostrokach**
Added a binary for the program `knot`. It is useful in validating protein homology models, since "bad" models are more likely to be knotted than "good" models.

### CODE REVIEW

File: recipes/knot/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  build:
+    - system
+  run:
+    - system
+
+test:
+  requires: 
+    - knot
+  commands:
+    - which knot
```
Comment (johanneskoester) [inline recipes/knot/meta.yaml]: Can you rather test if the command itself works? E.g. with --version or --help?

File: recipes/knot/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+requirements:
+  build:
+    - system
+  run:
+    - system
+
+test:
+  requires: 
+    - knot
```
Comment (johanneskoester) [inline recipes/knot/meta.yaml]: The package itself does not need to be added as dependency.

File: recipes/knot/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  fn: KNOT-1.0.0.tgz
+  url: http://mathbio.nimr.mrc.ac.uk/download/KNOT/KNOT-1.0.0.tgz
+  md5: 2556876109bae506fe171a45902db0b6
+
+build:
+  number: 1
+
+requirements:
+  build:
+    - system
```
Comment (johanneskoester) [inline recipes/knot/meta.yaml]: Can you comment on this dependency? I think if there are no real dependencies, you can also just skip the whole section.

File: recipes/knot/meta.yaml (hunk 1 above)
Comment (ostrokach) [inline recipes/knot/meta.yaml]: `knot` does not have `--version` or `--help`. I could download a PDB file and see that it parses it correctly. Do you think it's worth it?

File: recipes/knot/meta.yaml (hunk 3 above)
Comment (ostrokach) [inline recipes/knot/meta.yaml]: You are right. I checked the `system` package, and it's definitely not what I thought it was. My reason for including it was that the `${PREFIX}/bin` and `${PREFIX}/log` folders don't get created automatically if there are no dependencies. However, that's better fixed with `mkdir -p`.

File: recipes/knot/meta.yaml (hunk 2 above)
Comment (ostrokach) [inline recipes/knot/meta.yaml]: You are right. I had problems with this before, but it seems to work OK now.

File: recipes/knot/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+  fn: KNOT-1.0.0.tgz
+  url: http://mathbio.nimr.mrc.ac.uk/download/KNOT/KNOT-1.0.0.tgz
+  md5: 2556876109bae506fe171a45902db0b6
+
+build:
+  number: 1
+
+test:
+  commands:
+    - which knot
```
Comment (johanneskoester) [inline recipes/knot/meta.yaml]: Can you use knot --help or something similar here, in order to ensure that the binary really works?

Comment (johanneskoester) [inline recipes/knot/meta.yaml]: Sorry, I didn't see your comment. What about running it without args and use grep to check for an expected error message.

Comment (ostrokach) [inline recipes/knot/meta.yaml]: Running it without args throws a `Sementation Fault` with return code 139.

Comment (johanneskoester) [inline recipes/knot/meta.yaml]: Wow, what a shitty tool :-). Ok, then let's keep it as it is for now.

**Pull request merged**